src/Rules/Methods/FriendMethodCallRule.php

```diff
@@ -0,0 +1,93 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Rules\Methods;
+
+use PhpParser\Node;
+use PhpParser\Node\Expr\MethodCall;
+use PHPStan\Analyser\Scope;
+use PHPStan\PhpDoc\Tag\FriendTag;
+use PHPStan\Rules\RuleErrorBuilder;
+use PHPStan\Type\FileTypeMapper;
+use PHPStan\Type\TypeWithClassName;
+
+/**
+ * @implements \PHPStan\Rules\Rule<\PhpParser\Node\Expr\MethodCall>
+ */
+class FriendMethodCallRule implements \PHPStan\Rules\Rule
+{
+
+	private FileTypeMapper $fileTypeMapper;
+
+	public function __construct(FileTypeMapper $fileTypeMapper)
+	{
+		$this->fileTypeMapper = $fileTypeMapper;
+	}
+
+	public function getNodeType(): string
+	{
+		return MethodCall::class;
+	}
+
+	public function processNode(Node $node, Scope $scope): array
+	{
+		if (!$node->name instanceof Node\Identifier) {
+			return [];
+		}
+		$methodName = $node->name->name;
+		$callee = $scope->getType($node->var);
+		if (!$callee->hasMethod($methodName)->yes()) {
+			return [];
+		}
+		$method = $callee->getMethod($methodName, $scope);
+		if (!$scope->canCallMethod($method)) {
+			return [];
+		}
+		$docComment = $method->getDocComment();
+		if ($docComment === null) {
+			return [];
+		}
+		$friends = $this->fileTypeMapper->getResolvedPhpDoc(
+			$method->getDeclaringClass()->getFileName(),
+			$method->getDeclaringClass()->getName(),
+			null,
+			$method->getName(),
+			$docComment
+		)->getFriends();
+		$friends = array_filter($friends, static function (FriendTag $friend): bool {
+			return $friend->getType() instanceof TypeWithClassName;
+		});
+		if (\count($friends) === 0) {
+			return [];
+		}
+
+		if (!$scope->isInClass()) {
+			// no support for being friends with something that is not in a class
```

Comment:
Arguably these methods shouldn't be called outside of classes at all so this should be reported?

Reply:
Will be :)